File: 24-simple-starter/src/main/java/com/github/dnvriend/Application.java
package com.github.dnvriend;

import com.dnvriend.starter.Greeter;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class Application implements CommandLineRunner {
    private final Greeter greeter;

    public Application(Greeter greeter) {
        this.greeter = greeter;
    }

    @Override
    public void run(String... args) throws Exception {
        System.out.println(greeter.greet());
    }

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}

File: 25-simple-scala/src/main/scala/com/github/dnvriend/Application.scala
package com.github.dnvriend

import com.typesafe.scalalogging.LazyLogging
import org.springframework.boot.{CommandLineRunner, SpringApplication}
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.stereotype.Component

@SpringBootApplication
class Application

object Application {
  def main(args: Array[String]): Unit = {
      SpringApplication.run(classOf[Application], args:_*)
  }
}

@Component
class MyConsoleRunner extends CommandLineRunner with LazyLogging {
  override def run(args: String*): Unit = {
    logger.debug("This is very convenient ;-)")
  }
}